src/caldav/src/SubscriptionPlugin.php

```
@@ -0,0 +1,156 @@
+        }
+
+        // Block CS:share POST on subscription calendars so users can't
+        // manually re-share or change access levels.
+        $server->on('method:POST', [$this, 'blockSharePost'], 80);
+    }
+
+    /**
+     * Reject any write verb that targets a subscription calendar or one
+     * of its objects. Returns null when the path is unrelated.
+     */
+    public function blockWrite(RequestInterface $request, ResponseInterface $response)
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$response'. (undefined)

(UnusedFormalParameter)